include/sparrow/array_data_concepts.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "sparrow/dictionary_encoded_layout.hpp"
+#include "sparrow/fixed_size_layout.hpp"
+#include "sparrow/mp_utils.hpp"
+#include "sparrow/variable_size_binary_layout.hpp"
+
+namespace sparrow
+{
+    /**
+     * Concept to check if a layout is a supported layout.
+     *
+     * A layout is considered supported if it is an instance of `fixed_size_layout`,
+     * `variable_size_binary_layout`, or `dictionary_encoded_layout`.
+     *
+     * @tparam Layout The layout type to check.
+     */
+    template <class Layout>
+    concept is_a_supported_layout = mpl::is_type_instance_of_v<Layout, fixed_size_layout>
```

Comment:
Note that concepts with 1 parameter can be used like this:
```
template< myconcept T >
...

// and
void somefunc( auto myconcept T ) { ... }
```

so with a few exception it might be better to name this `supported_layout` or some similar name, instead of is_* name. Not a big issue at the moment, we can change it later.

(after some thinking)
How about `arrow_layout`?